plugins/processors/enum/enum.go

```diff
@@ -37,12 +39,16 @@ func (mapper *EnumMapper) Init() error {
 	mapper.FieldFilters = make(map[string]filter.Filter)
 	mapper.TagFilters = make(map[string]filter.Filter)
 	for _, mapping := range mapper.Mappings {
+		// Handle deprecated field option
 		if mapping.Field != "" {
-			fieldFilter, err := filter.NewIncludeExcludeFilter([]string{mapping.Field}, nil)
+			mapping.Fields = append(mapping.Fields, mapping.Field)
+		}
+		for _, field := range mapping.Fields {
+			fieldFilter, err := filter.NewIncludeExcludeFilter([]string{field}, nil)
 			if err != nil {
 				return fmt.Errorf("failed to create new field filter: %w", err)
 			}
-			mapper.FieldFilters[mapping.Field] = fieldFilter
+			mapper.FieldFilters[field] = fieldFilter
 		}
```

Comment:
I'm a bit puzzled here. Why don't you just create the filter using _all_ fields? That's what filters are here for...  Any reason not to do

```golang
		fieldFilter, err := filter.Compile(range mapping.Fields)
		if err != nil {
			return fmt.Errorf("failed to create new field filter: %w", err)
		}
		mapper.FieldFilter = fieldFilter
```
and then later just call `mapper.FieldFilter.Match(key)`...

Reply:
I agree this is a much cleaner approach, I'll change it to this